src/cursor.c

```diff
@@ -891,22 +913,32 @@ handle_cursor_axis(struct server *server, struct cursor_context *ctx,
 	uint32_t modifiers = wlr_keyboard_get_modifiers(
 			&server->seat.keyboard_group->keyboard);
 
-	uint32_t button = 0;
-	if (event->delta_discrete < 0) {
-		button = BTN_GEAR_UP;
-	} else if (event->delta_discrete > 0) {
-		button = BTN_GEAR_DOWN;
+	enum direction direction = LAB_DIRECTION_INVALID;
+	if (event->orientation == WLR_AXIS_ORIENTATION_HORIZONTAL) {
+		int rel = compare_delta(event, &server->seat.smooth_scroll_offset.x);
+		if (rel < 0) {
+			direction = LAB_DIRECTION_LEFT;
+		} else if (rel > 0) {
+			direction = LAB_DIRECTION_RIGHT;
+		}
+	} else if (event->orientation == WLR_AXIS_ORIENTATION_VERTICAL) {
+		int rel = compare_delta(event, &server->seat.smooth_scroll_offset.y);
+		if (rel < 0) {
+			direction = LAB_DIRECTION_UP;
+		} else if (rel > 0) {
+			direction = LAB_DIRECTION_DOWN;
+		}
+	} else {
+		wlr_log(WLR_DEBUG, "Failed to handle cursor axis event");
 	}
 
-	if (!button || event->source != WLR_AXIS_SOURCE_WHEEL
-			|| event->orientation != WLR_AXIS_ORIENTATION_VERTICAL) {
-		wlr_log(WLR_DEBUG, "Failed to handle cursor axis event");
+	if (direction == LAB_DIRECTION_INVALID) {
 		return false;
 	}
 
 	wl_list_for_each(mousebind, &rc.mousebinds, link) {
 		if (ssd_part_contains(mousebind->context, ctx->type)
-				&& mousebind->button == button
+				&& mousebind->direction == direction
 				&& modifiers == mousebind->modifiers
 				&& mousebind->mouse_event == MOUSE_ACTION_SCROLL) {
 			handled = true;
```

Comment:
Do we want to reset the accumulator after executing a binding?

Reply:
I don't think we do--when the user stops scrolling, the accumulator is reset, and if the user continues scrolling, we don't want to drop the additional partial progress they may have made towards the next quantized activation.